integration-tests/tests/circuits.rs

```diff
@@ -88,23 +68,51 @@ async fn test_state_circuit_block(block_num: u64) {
     prover.verify().expect("state_circuit verification failed");
 }
 
-#[tokio::test]
-async fn test_state_circuit_block_transfer_0() {
-    log_init();
-    let block_num = GEN_DATA.blocks.get("Transfer 0").unwrap();
-    test_state_circuit_block(*block_num).await;
+macro_rules! declare_tests {
+    ($test_evm_name:ident, $test_state_name:ident, $block_tag:expr) => {
+        #[tokio::test]
+        async fn $test_evm_name() {
+            log_init();
+            let block_num = GEN_DATA.blocks.get($block_tag).unwrap();
+            test_evm_circuit_block(*block_num).await;
+        }
+
+        #[tokio::test]
+        async fn $test_state_name() {
+            log_init();
+            let block_num = GEN_DATA.blocks.get($block_tag).unwrap();
+            test_state_circuit_block(*block_num).await;
+        }
+    };
```

Comment:
When I wrote the first integration tests I looked through some library to do this but I decided to settle with doing the setup manually because it's just a single line: `log_init();`.  And there's no teardown. So by using for example the `test-context` we'd be replacing the line `log_init();` by another line `#[test_context(MyAsyncContext)]`, so I think there's no much gain!  Also the setup data in our case is static (the `GEN_DATA`), so it can live under a `lazy_static` and that already works fine :)

But thanks for the suggestion!